drivers/spi/spi_nrfx_spim.c

```diff
@@ -480,140 +507,128 @@ static int spim_nrfx_pm_action(const struct device *dev,
-		 .miso_pin  = DT_PROP_OR(SPIM(idx), miso_pin,		\
-					 NRFX_SPIM_PIN_NOT_USED),	\
-		 .miso_pull = SPIM_NRFX_MISO_PULL(idx),))
-
-#define SPI_NRFX_SPIM_DEVICE(idx)					       \
-	NRF_DT_CHECK_PIN_ASSIGNMENTS(SPIM(idx), 1,			       \
+#define SPI_NRFX_SPIM_EXTENDED_CONFIG(inst)				       \
+	IF_ENABLED(NRFX_SPIM_EXTENDED_ENABLED,				       \
+		   (.dcx_pin = NRFX_SPIM_PIN_NOT_USED,			       \
+		    IF_ENABLED(DT_INST_NODE_HAS_PROP(inst, has_rx_delay),      \
+			       (.rx_delay = DT_INST_PROP(inst, rx_delay),))    \
+		   ))
+
+#define SPI_NRFX_SPIM_PIN_CFG(inst)					       \
+	COND_CODE_1(CONFIG_PINCTRL,					       \
+		    (.skip_gpio_cfg = true,				       \
+		     .skip_psel_cfg = true,),				       \
+		    (.sck_pin   = DT_INST_PROP(inst, sck_pin),		       \
+		     .mosi_pin  = DT_INST_PROP_OR(inst, mosi_pin,	       \
+						  NRFX_SPIM_PIN_NOT_USED),     \
+		     .miso_pin  = DT_INST_PROP_OR(inst, miso_pin,	       \
+						  NRFX_SPIM_PIN_NOT_USED),     \
+		     .miso_pull = SPIM_NRFX_MISO_PULL(inst),))
+
+#define SPIM_MEMORY_SECTION(inst)					       \
+	IF_ENABLED(DT_INST_NODE_HAS_PROP(inst, memory_regions),		       \
+		   (__attribute__((__section__(LINKER_DT_NODE_REGION_NAME(     \
+			DT_INST_PHANDLE(inst, memory_regions)))))))
+
+#define SPI_NRFX_SPIM_DEFINE(inst)					       \
+	NRF_DT_CHECK_PIN_ASSIGNMENTS(DT_DRV_INST(inst), 1,		       \
 				     sck_pin, mosi_pin, miso_pin);	       \
 	BUILD_ASSERT(IS_ENABLED(CONFIG_PINCTRL) ||			       \
-		     !(SPIM_PROP(idx, miso_pull_up) &&			       \
-		       SPIM_PROP(idx, miso_pull_down)),			       \
-		"SPIM"#idx						       \
+		     !(DT_INST_PROP(inst, miso_pull_up) &&		       \
+		       DT_INST_PROP(inst, miso_pull_down)),		       \
+		"SPIM"#inst						       \
 		": cannot enable both pull-up and pull-down on MISO line");    \
-	static int spi_##idx##_init(const struct device *dev)		       \
+									       \
+	static void irq_connect##inst(void)				       \
 	{								       \
-		struct spi_nrfx_data *dev_data = dev->data;		       \
-		int err;						       \
-		IRQ_CONNECT(NRFX_IRQ_NUMBER_GET(NRF_SPIM##idx),		       \
-			    DT_IRQ(SPIM(idx), priority),		       \
-			    nrfx_isr, nrfx_spim_##idx##_irq_handler, 0);       \
-		IF_ENABLED(CONFIG_PINCTRL, (				       \
-			const struct spi_nrfx_config *dev_config = dev->config;\
-			err = pinctrl_apply_state(dev_config->pcfg,	       \
-						  PINCTRL_STATE_DEFAULT);      \
-			if (err < 0) {					       \
-				return err;				       \
-			}						       \
-		))							       \
-		err = spi_context_cs_configure_all(&dev_data->ctx);	       \
-		if (err < 0) {						       \
-			return err;					       \
-		}							       \
-		spi_context_unlock_unconditionally(&dev_data->ctx);	       \
-		COND_CODE_1(CONFIG_SOC_NRF52832_ALLOW_SPIM_DESPITE_PAN_58,     \
-		(return anomaly_58_workaround_init(dev);),		       \
-		(return 0;))						       \
+		IRQ_CONNECT(DT_INST_IRQN(inst), DT_INST_IRQ(inst, priority),   \
```

Comment:
Even if we don't use everything in this PR, we need to keep this. We are currently ignoring the IRQ number in the .dts(i) files, and that is a problem. The same goes for `NRFX_SPIM_INSTANCE`.